packages/visualizations/src/components/Map/MapRender.svelte

```diff
@@ -129,23 +133,34 @@
     function sourceLoadingCallback(e: MapSourceDataEvent) {
         // sourceDataType can be "visibility" or "metadata", in which case it's not about the data itself
         if (e.isSourceLoaded && e.sourceId === sourceId && !e.sourceDataType) {
-            // The type forces you to pass a filter parameter in the option, but it's not required by the real code
-            // https://github.com/maplibre/maplibre-gl-js/issues/1393
-            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-            // @ts-ignore
-            const renderedFeatures = map.querySourceFeatures(sourceId, { sourceLayer: layerId });
-
+            const renderedFeatures = map.querySourceFeatures(
+                sourceId,
+                // The type forces you to pass a filter parameter in the option, but it's not required by the real code
+                // https://github.com/maplibre/maplibre-gl-js/issues/1393
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                {
+                    sourceLayer: layer['source-layer'] || layerId, // FIXME: This may not the best way to do it
+                }
+            );
             if (renderedFeatures.length) {
-                // Restrict zoom max
+                // Restrict zoom max only works for geojson for now
                 // TODO: We may not catch the smaller shapes if Maplibre discarded them for rendering reasons, so it's a bit risky. Is it worth it?
                 // A low-cost approach could be to restrict the zoom scale to an arbitrary value (e.g. only 4 from the max zoom)... or not restrict at all.
-                const maxZoom = computeMaxZoomFromGeoJsonFeatures(container, renderedFeatures);
-                map.setMaxZoom(maxZoom);
+                if (e.source.type === 'geojson') {
```

Comment:
I'd be tempted to also support activeShapes only for GeoJSON sources. I feel like it's going to be very wonky. Then the entire `sourceLoadingCallback` (and therefore the fact that we listen to `sourcedata` in the first place) probably only makes sense for GeoJSON.

Reply:
Since we're using activeShapes for test, I just added a warning in the comments, and we may even want to not document that at all in the future. But it's useful for us to keep it for now indeed.